libvast/builtins/connectors/file.cpp

```diff
@@ -0,0 +1,268 @@
+            if (current_data.empty()) {
+              if (not eof_reached) {
+                co_yield chunk::make_empty();
+                continue;
+              }
+              if (eof_reached and not following) {
+                break;
+              }
+            }
+          }
+          if (eof_reached or current_data.size() == max_chunk_size) {
+            auto chunk = chunk::make(std::exchange(current_data, {}));
+            co_yield std::move(chunk);
+            if (following or not eof_reached) {
+              current_data.reserve(max_chunk_size);
+            }
+          }
+        }
+        if (fd != STDIN_FILENO) {
+          ::close(fd);
+        }
+        co_return;
+      },
+      read_timeout, fd, following);
+  }
+
+  auto default_parser(std::span<std::string const> args) const
+    -> std::pair<std::string, std::vector<std::string>> override {
+    for (std::string_view arg : args) {
+      if (arg == "-" || arg == "stdin") {
+        break;
+      }
+      return {detail::file_path_to_parser(arg), {}};
+    }
+    return {"json", {}};
+  }
+
+  auto initialize(const record&, const record& global_config)
+    -> caf::error override {
+    if (!global_config.contains("vast")) {
+      return caf::none;
+    }
+    const auto* vast_settings = caf::get_if<record>(&global_config.at("vast"));
+    if (!vast_settings || !vast_settings->contains("import")) {
+      return caf::none;
+    }
+    const auto* import_settings
+      = caf::get_if<record>(&vast_settings->at("import"));
+    if (!import_settings || !import_settings->contains("read-timeout")) {
+      return caf::none;
+    }
+    const auto* read_timeout_entry
+      = caf::get_if<std::string>(&import_settings->at("read-timeout"));
+    if (!read_timeout_entry) {
+      return caf::none;
+    }
+    if (auto timeout_duration = to<vast::duration>(*read_timeout_entry)) {
+      read_timeout_ = std::chrono::duration_cast<std::chrono::milliseconds>(
+        *timeout_duration);
+    }
+    return caf::none;
+  }
+
+  auto name() const -> std::string override {
+    return "file";
+  }
+
+private:
+  std::chrono::milliseconds read_timeout_{vast::defaults::import::read_timeout};
+};
+
+} // namespace vast::plugins::file
+
+namespace vast::plugins::stdin_ {
+
+class plugin : public virtual vast::plugins::file::plugin {
+public:
+  auto make_loader([[maybe_unused]] std::span<std::string const> args,
+                   operator_control_plane& ctrl) const
+    -> caf::expected<generator<chunk_ptr>> override {
+    std::vector<std::string> new_args = {args.begin(), args.end()};
+    new_args.emplace_back("-");
+    return vast::plugins::file::plugin::make_loader(new_args, ctrl);
+  }
+
+  auto default_parser([[maybe_unused]] std::span<std::string const> args) const
+    -> std::pair<std::string, std::vector<std::string>> override {
+    return {"json", {}};
+  }
+
+  auto name() const -> std::string override {
+    return "stdin";
+  }
```

Comment:
Related: We would now have two plugins with the same name.

Reply:
I removed `stdin.cpp` in https://github.com/tenzir/vast/pull/3088/commits/305e4c863f71b98e8bc9cba3db8e7da98eeb11b2